addons/html_editor/static/src/plugin.js

```diff
@@ -42,16 +42,39 @@ export class Plugin {
         return !isProtecting(ev.target) && (!isProtected(ev.target) || isUnprotecting(ev.target));
     }
 
-    addDomListener(target, eventName, fn, capture = false) {
+    /**
+     * Add an event listener on a given target, that will only be executed if
+     * the target is valid (unless `isGlobal` is true), and ensure it is removed
+     * when we destroy the editor.
+     *
+     * @param {Element} target
+     * @param {string} eventName
+     * @param {function(Event):void} fn
+     * @param {boolean} [capture=false] `useCapture` flag of `addEventListener`
+     * @param {boolean} [isGlobal=false] if true, don't check target validity
+     */
+    addDomListener(target, eventName, fn, capture = false, isGlobal = false) {
         const handler = (ev) => {
-            if (this.isValidTargetForDomListener(ev)) {
+            if (isGlobal || this.isValidTargetForDomListener(ev)) {
                 fn?.call(this, ev);
             }
         };
         target.addEventListener(eventName, handler, capture);
         this._cleanups.push(() => target.removeEventListener(eventName, handler, capture));
     }
 
+    /**
+     * Add an event listener on the editor's document, and ensure it is removed
+     * when we destroy the editor.
+     *
+     * @param {string} eventName
+     * @param {function(Event):void} fn
+     * @param {boolean} [capture=false] `useCapture` flag of `addEventListener`
+     */
+    addGlobalDomListener(eventName, fn, capture = false) {
```

Comment:
I'm not a big far of the `global` labeling.
In view of the issue we have when the editor is setup in an ifram( website) people could deduce that the `addGlobalDomListener` will listen to even on the main page document, and not the iframe document.

maybe we could name it `addDocumentDomListener` ?

Reply:
I see, could you just add a `// @todo use this function to avoid iframe problems` or something to that effect.  
In order to clarify that it is build _also_ as a future potential "fix" for iframe problems.